packages/codegen/src/codegen.ts

```diff
@@ -0,0 +1,131 @@
+import { codegen } from '@graphql-codegen/core';
+import { preset } from '@graphql-codegen/near-operation-file-preset';
+import * as typedDocumentNodePlugin from '@graphql-codegen/typed-document-node';
+import * as typescriptOperationsPlugin from '@graphql-codegen/typescript-operations';
+import { printExecutableGraphQLDocument } from '@graphql-tools/documents';
+import { readFile, writeFile } from 'fs/promises';
+import { globby } from 'globby';
+import type { DocumentNode } from 'graphql';
+import { parse } from 'graphql';
+import * as path from 'node:path';
+import { defaultHash } from './default-hash.js';
+import { generateDocumentFiles } from './generate-document-files.js';
+import { generateSchemaTypes } from './generate-schema-types.js';
+import type { CodegenArgs, OutputFile } from './types.js';
+import { changeExtension } from './utils.js';
+
+export async function athenaCodegen({
+  schemaName,
+  documents,
+  baseDir,
+  extension,
+  hash,
+}: CodegenArgs): Promise<void> {
+  const outputFiles: Array<OutputFile> = [];
+  const persistedQueries: Record<string, string> = {};
+  const schemaPath = path.join(baseDir, schemaName);
+  // Read schema
+  const rawSchema = await readFile(schemaPath, 'utf-8');
+  const parsedSchema = parse(rawSchema);
+  const hashFn = hash || defaultHash;
+
+  // Generate schema types and write to root schema file
+  const schemaTypesOutputPath = changeExtension(schemaPath, extension);
+  const schemaTypes = await generateSchemaTypes(
+    parsedSchema,
+    schemaTypesOutputPath
+  );
+
+  outputFiles.push(schemaTypes);
+
+  // Expand glob patterns and find matching files
+  const paths = await globby([...documents, `!${schemaPath}`]);
+
+  // Create DocumentFile objects for use in the preset builder
+  const documentFiles = await generateDocumentFiles(paths);
+
+  function onExecutableDocumentNode(documentNode: DocumentNode) {
+    const materializedDocumentString =
+      printExecutableGraphQLDocument(documentNode);
+
+    const queryId = hashFn(documentNode);
+
+    persistedQueries[queryId] = materializedDocumentString;
+
+    if (process.env.NODE_ENV === 'prod') {
+      return {
+        queryId,
+      };
+    } else {
+      return { queryId, $DEBUG: { contents: materializedDocumentString } };
+    }
+  }
+
+  const configs = await preset.buildGeneratesSection({
+    baseOutputDir: baseDir,
+    config: {
+      useTypeImports: true,
+    },
+    presetConfig: {
+      baseTypesPath: path.relative(baseDir, schemaTypesOutputPath),
+      extension,
+    },
+    schema: parsedSchema,
+    documents: documentFiles,
```

Comment:
a stupid (but easy + quick) thing we might do to mitigate the fragment de-duping is run this for each document.

Reply:
I ended up just handling this by removing the weird `_0` suffix before writing out the contents to a file (in a different, forthcoming PR)